### PR TITLE
refact(charts, operator): added labels to make helm and operator uniform

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

**What this PR does / why we need it:**
This PR adds deployment labels which are present in operator but not in  helm charts.

**helm upgrade result:**
```
user:openebs$ helm install --namespace openebs --name openebs stable/openebs --version 1.0.0
NAME:   openebs
LAST DEPLOYED: Mon Jul 15 10:56:13 2019
NAMESPACE: openebs
STATUS: DEPLOYED

RESOURCES:
==> v1/ClusterRole
NAME     AGE
openebs  4s

==> v1/ClusterRoleBinding
NAME     AGE
openebs  4s

==> v1/ConfigMap
NAME                DATA  AGE
openebs-ndm-config  1     4s

==> v1/Deployment
NAME                         READY  UP-TO-DATE  AVAILABLE  AGE
openebs-admission-server     0/1    1           0          3s
openebs-apiserver            0/1    1           0          3s
openebs-localpv-provisioner  0/1    1           0          3s
openebs-ndm-operator         0/1    1           0          3s
openebs-provisioner          0/1    1           0          3s
openebs-snapshot-operator    0/1    1           0          3s

==> v1/Pod(related)
NAME                                          READY  STATUS             RESTARTS  AGE
openebs-admission-server-74c7f77cff-n6llb     0/1    ContainerCreating  0         3s
openebs-apiserver-7fc86cc574-lj5b9            0/1    ContainerCreating  0         3s
openebs-localpv-provisioner-7879ccd58b-zjmnz  0/1    ContainerCreating  0         3s
openebs-ndm-bf7ld                             0/1    ContainerCreating  0         3s
openebs-ndm-operator-74bcd75cfd-vd4js         0/1    Pending            0         3s
openebs-provisioner-84f98747f-l4qld           0/1    ContainerCreating  0         3s
openebs-snapshot-operator-67c9b4c4bf-xhbpt    0/2    ContainerCreating  0         3s

==> v1/Secret
NAME                    TYPE    DATA  AGE
admission-server-certs  Opaque  2     4s

==> v1/Service
NAME                  TYPE       CLUSTER-IP   EXTERNAL-IP  PORT(S)   AGE
admission-server-svc  ClusterIP  10.0.12.147  <none>       443/TCP   4s
openebs-apiservice    ClusterIP  10.0.9.54    <none>       5656/TCP  3s

==> v1/ServiceAccount
NAME     SECRETS  AGE
openebs  1        4s

==> v1beta1/DaemonSet
NAME         DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
openebs-ndm  1        1        0      1           0          <none>         3s

==> v1beta1/ValidatingWebhookConfiguration
NAME                            AGE
openebs-validation-webhook-cfg  3s


NOTES:
The OpenEBS has been installed. Check its status by running:
$ kubectl get pods -n openebs

For dynamically creating OpenEBS Volumes, you can either create a new StorageClass or
use one of the default storage classes provided by OpenEBS.

Use `kubectl get sc` to see the list of installed OpenEBS StorageClasses. A sample
PVC spec using `openebs-jiva-default` StorageClass is given below:"

---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: demo-vol-claim
spec:
  storageClassName: openebs-jiva-default
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 5G
---

For more information, please visit http://docs.openebs.io/.

Please note that, OpenEBS uses iSCSI for connecting applications with the
OpenEBS Volumes and your nodes should have the iSCSI initiator installed.


user:openebs$ kubectl get pods -n openebs
NAME                                           READY   STATUS    RESTARTS   AGE
openebs-admission-server-74c7f77cff-n6llb      1/1     Running   0          35s
openebs-apiserver-7fc86cc574-lj5b9             1/1     Running   0          35s
openebs-localpv-provisioner-7879ccd58b-zjmnz   1/1     Running   0          35s
openebs-ndm-bf7ld                              1/1     Running   0          35s
openebs-ndm-operator-74bcd75cfd-vd4js          1/1     Running   0          35s
openebs-provisioner-84f98747f-l4qld            1/1     Running   0          35s
openebs-snapshot-operator-67c9b4c4bf-xhbpt     2/2     Running   0          35s
user:openebs$ pwd
/home/user/work/src/github.com/openebs/openebs/k8s/charts/openebs
user:openebs$ helm upgrade openebs .
Release "openebs" has been upgraded.
LAST DEPLOYED: Mon Jul 15 10:58:24 2019
NAMESPACE: openebs
STATUS: DEPLOYED

RESOURCES:
==> v1/ClusterRole
NAME     AGE
openebs  2m14s

==> v1/ClusterRoleBinding
NAME     AGE
openebs  2m14s

==> v1/ConfigMap
NAME                DATA  AGE
openebs-ndm-config  1     2m14s

==> v1/Deployment
NAME                         READY  UP-TO-DATE  AVAILABLE  AGE
openebs-admission-server     1/1    1           1          2m13s
openebs-apiserver            1/1    1           1          2m13s
openebs-localpv-provisioner  1/1    1           1          2m13s
openebs-ndm-operator         1/1    1           1          2m13s
openebs-provisioner          1/1    1           1          2m13s
openebs-snapshot-operator    1/1    1           1          2m13s

==> v1/Pod(related)
NAME                                          READY  STATUS       RESTARTS  AGE
openebs-admission-server-74c7f77cff-n6llb     1/1    Running      0         2m13s
openebs-apiserver-7fc86cc574-lj5b9            1/1    Running      0         2m13s
openebs-localpv-provisioner-7879ccd58b-zjmnz  1/1    Terminating  0         2m13s
openebs-localpv-provisioner-7d7b4cc56d-6rh2h  1/1    Running      0         2s
openebs-ndm-bf7ld                             1/1    Running      0         2m13s
openebs-ndm-operator-74bcd75cfd-vd4js         1/1    Running      0         2m13s
openebs-provisioner-84f98747f-l4qld           1/1    Running      0         2m13s
openebs-snapshot-operator-67c9b4c4bf-xhbpt    2/2    Running      0         2m13s

==> v1/Secret
NAME                    TYPE    DATA  AGE
admission-server-certs  Opaque  2     2m14s

==> v1/Service
NAME                  TYPE       CLUSTER-IP   EXTERNAL-IP  PORT(S)   AGE
admission-server-svc  ClusterIP  10.0.12.147  <none>       443/TCP   2m14s
openebs-apiservice    ClusterIP  10.0.9.54    <none>       5656/TCP  2m13s

==> v1/ServiceAccount
NAME     SECRETS  AGE
openebs  1        2m14s

==> v1beta1/DaemonSet
NAME         DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
openebs-ndm  1        1        1      1           1          <none>         2m13s

==> v1beta1/ValidatingWebhookConfiguration
NAME                            AGE
openebs-validation-webhook-cfg  2m13s


NOTES:
The OpenEBS has been installed. Check its status by running:
$ kubectl get pods -n openebs

For dynamically creating OpenEBS Volumes, you can either create a new StorageClass or
use one of the default storage classes provided by OpenEBS.

Use `kubectl get sc` to see the list of installed OpenEBS StorageClasses. A sample
PVC spec using `openebs-jiva-default` StorageClass is given below:"

---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: demo-vol-claim
spec:
  storageClassName: openebs-jiva-default
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 5G
---

For more information, please visit http://docs.openebs.io/.

Please note that, OpenEBS uses iSCSI for connecting applications with the
OpenEBS Volumes and your nodes should have the iSCSI initiator installed.


user:openebs$ kubectl get pods -n openebs --show-labels 
NAME                                           READY   STATUS    RESTARTS   AGE     LABELS
openebs-admission-server-74c7f77cff-n6llb      1/1     Running   0          3m59s   app=admission-webhook,name=admission-webhook,openebs.io/component-name=admission-webhook,openebs.io/version=1.0.0,pod-template-hash=74c7f77cff
openebs-apiserver-7fc86cc574-lj5b9             1/1     Running   0          3m59s   app=openebs,component=apiserver,name=maya-apiserver,openebs.io/component-name=maya-apiserver,openebs.io/version=1.0.0,pod-template-hash=7fc86cc574,release=openebs
openebs-localpv-provisioner-7d7b4cc56d-6rh2h   1/1     Running   0          108s    app=openebs,component=localpv-provisioner,name=openebs-localpv-provisioner,openebs.io/component-name=openebs-localpv-provisioner,openebs.io/version=1.0.0,pod-template-hash=7d7b4cc56d,release=openebs
openebs-ndm-bf7ld                              1/1     Running   0          3m59s   app=openebs,component=ndm,controller-revision-hash=5cc78dd4b4,name=openebs-ndm,openebs.io/component-name=ndm,openebs.io/version=1.0.0,pod-template-generation=1,release=openebs
openebs-ndm-operator-74bcd75cfd-vd4js          1/1     Running   0          3m59s   app=openebs,component=ndm-operator,name=ndm-operator,openebs.io/component-name=ndm-operator,openebs.io/version=1.0.0,pod-template-hash=74bcd75cfd,release=openebs
openebs-provisioner-84f98747f-l4qld            1/1     Running   0          3m59s   app=openebs,component=provisioner,name=openebs-provisioner,openebs.io/component-name=openebs-provisioner,openebs.io/version=1.0.0,pod-template-hash=84f98747f,release=openebs
openebs-snapshot-operator-67c9b4c4bf-xhbpt     2/2     Running   0          3m59s   app=openebs,component=snapshot-operator,name=openebs-snapshot-operator,openebs.io/component-name=openebs-snapshot-operator,openebs.io/version=1.0.0,pod-template-hash=67c9b4c4bf,release=openebs

```

### CODE REVIEW

File: k8s/charts/openebs/templates/deployment-admission-server.yaml [hunk 1]
```diff
@@ -8,8 +8,8 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
     component: admission-webhook
-    name: admission-webhook
-    openebs.io/component-name: maya-apiserver
+    openebs.io/component-name: admission-server
```
Comment (kmova) [inline k8s/charts/openebs/templates/deployment-admission-server.yaml]: everyone -- admission-webhook is being used, why change to admission-server? Should we instead change the operator.yaml to have the name as admission-webhook ?

File: k8s/charts/openebs/templates/deployment-local-provisioner.yaml [hunk 2]
```diff
@@ -54,7 +57,7 @@ spec:
         - name: OPENEBS_IO_BASE_PATH
           value: "{{ .Values.localprovisioner.basePath }}"
         - name: OPENEBS_IO_HELPER_IMAGE
-          value: "{{ .Values.localprovisioner.helperImage }}"
+          value: "{{ .Values.localprovisioner.helperImage }}:{{ .Values.localprovisioner.helperImageTag }}"
```
Comment (kmova) [inline k8s/charts/openebs/templates/deployment-local-provisioner.yaml]: Wow! good catch!

File: k8s/charts/openebs/templates/validationwebhook.yaml [hunk 3]
```diff
@@ -42,13 +42,6 @@ metadata:
     chart: {{ template "openebs.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
-  # Helm hook annotations in order to ensure that the certs
```
Comment (kmova) [inline k8s/charts/openebs/templates/validationwebhook.yaml]: Why is this removed?

Comment (shubham14bajpai) [reply k8s/charts/openebs/templates/validationwebhook.yaml]: Thus pre-install hook prevents the secret creation if someone uses the older version of charts and upgraded to new version.
[https://github.com/helm/charts/pull/14162](url)

File: k8s/charts/openebs/templates/deployment-admission-server.yaml (hunk 1 above)
Comment (shubham14bajpai) [reply k8s/charts/openebs/templates/deployment-admission-server.yaml]: Done

File: k8s/charts/openebs/templates/deployment-local-provisioner.yaml (hunk 2 above)
Comment (prateekpandey14) [inline k8s/charts/openebs/templates/deployment-local-provisioner.yaml]: Here `helperImage` also contains the tag info as well, but later we changed while raising the PR to upstream stable charts, please pull the values files as well from stable/charts.

File: k8s/charts/openebs/templates/deployment-admission-server.yaml [hunk 4]
```diff
@@ -8,8 +8,8 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
     component: admission-webhook
-    name: admission-webhook
-    openebs.io/component-name: maya-apiserver
```
Comment (prateekpandey14) [inline k8s/charts/openebs/templates/deployment-admission-server.yaml]: ahh good that we don't have this change in upstream :smile:

File: k8s/charts/openebs/templates/deployment-local-provisioner.yaml (hunk 2 above)
Comment (shubham14bajpai) [reply k8s/charts/openebs/templates/deployment-local-provisioner.yaml]: Done
